### PR TITLE
wireshark-qt: make wireshark-qt build and run again

### PR DESCRIPTION
**Pull request by exi**
wireshark-qt build is failing, this fixes it

### CODE REVIEW

File: pkgs/applications/networking/sniffers/wireshark/default.nix [hunk 1]
```diff
@@ -24,13 +25,19 @@ stdenv.mkDerivation {
...
 
   patches = [ ./wireshark-lookup-dumpcap-in-path.patch ];
 
+  makeFlags = [
+    "DESTDIR=$(out)"
+    "PREFIX="
+  ];
+
   configureFlags = "--disable-usr-local --disable-silent-rules --with-ssl"
+    + " --prefix=/"
```
Comment (peti) [inline pkgs/applications/networking/sniffers/wireshark/default.nix]: The `--prefix=/` flag here looks very suspicious. I have a hard time imagining that this setting configures a build that generates working binaries.

Comment (exi) [inline pkgs/applications/networking/sniffers/wireshark/default.nix]: I tested wireshark-qt and wireshark-gtk but i'll remove it and see whether it works without the --prefix.
Could very well be a leftover from experimentation. Thanks for the remark. Ill report back in 10 minutes.

File: pkgs/applications/networking/sniffers/wireshark/default.nix [hunk 2]
```diff
@@ -24,13 +25,19 @@ stdenv.mkDerivation {
...
   buildInputs = [
     bison flex perl pkgconfig libpcap lua5 openssl libgcrypt gnutls
-    geoip libnl c-ares python libcap glib
+    geoip libnl c-ares python libcap glib zlib
   ] ++ optional withQt qt4
     ++ optional withGtk gtk;
 
   patches = [ ./wireshark-lookup-dumpcap-in-path.patch ];
 
+  makeFlags = [
```
Comment (peti) [inline pkgs/applications/networking/sniffers/wireshark/default.nix]: The settings should be superfluous if the prefix is set properly at configure time.

File: pkgs/applications/networking/sniffers/wireshark/default.nix [hunk 3]
```diff
@@ -47,13 +54,23 @@ stdenv.mkDerivation {
...
   postInstall = optionalString (withQt || withGtk) ''
     mkdir -p "$out"/share/applications/
     mkdir -p "$out"/share/icons/
     cp "$desktopItem/share/applications/"* "$out/share/applications/"
     cp image/wsicon.svg "$out"/share/icons/wireshark.svg
-  '' + optionalString withQt ''
-    mv "$out/bin/wireshark-qt" "$out/bin/wireshark"
+  '';
+
+  postFixup = ''
```
Comment (peti) [inline pkgs/applications/networking/sniffers/wireshark/default.nix]: Ditto, these settings should be superfluous if the prefix is set properly at configure time.

File: pkgs/applications/networking/sniffers/wireshark/default.nix [hunk 4]
```diff
@@ -24,13 +25,19 @@ stdenv.mkDerivation {
 
   buildInputs = [
     bison flex perl pkgconfig libpcap lua5 openssl libgcrypt gnutls
-    geoip libnl c-ares python libcap glib
+    geoip libnl c-ares python libcap glib zlib
```
Comment (peti) [inline pkgs/applications/networking/sniffers/wireshark/default.nix]: Is `zlib` is added unconditionally here. Is that really necessary? Or is that dependency required only by the qt build variant, maybe?

Comment (exi) [inline pkgs/applications/networking/sniffers/wireshark/default.nix]: good question, the qt variant needs it, did not check gtk variant. Will get back to you on that.

**Comment by exi** [comment]
@peti you where right, i think i got a bit carried away with the PR.
I deleted most of the unnecessary code.
zlib is required by gtk and qt versions of wireshark.
the wireshark-qt binary seems to be called "wireshark".

I will debug some more as it seems that building wireshark-gtk from master results in a broken binary as well:
	libpangocairo-1.0.so.0 => not found
	libcairo.so.2 => not found
	libgdk_pixbuf-2.0.so.0 => not found
	libpango-1.0.so.0 => not found
I will adjust the build includes accordingly

**Comment by exi** [comment]
@peti i updated the PR with the necessary gtk dependencies. qt and gtk build not build and work for me

File: pkgs/applications/networking/sniffers/wireshark/default.nix [hunk 5]
```diff
@@ -24,9 +25,9 @@ stdenv.mkDerivation {
...
   ] ++ optional withQt qt4
-    ++ optional withGtk gtk;
+    ++ (optionals withGtk [gtk pango cairo gdk_pixbuf]);
```
Comment (peti) [inline pkgs/applications/networking/sniffers/wireshark/default.nix]: Minor nit: the `(...)` brackets around that expression are redundant and can be removed.

**Comment by peti** [comment]
The new patch looks much nicer than the first version! Thank you very much for your efforts.

**Pull request merged**